# **Usage**

## **Quick start**

To start your first analysis:

1. Copy your **paired-end** reads files, in **.fastq.gz** format, into: **./resources/reads/** directory

!!! note
   Without reads, SARS-CoV-2 from ```./resources/test\_data/``` directory will be used

2. Execute **Run_GeVarLi.sh** bash script to run GeVarLi pipeline:
    - or with a **Double-click** on it _(if you make .sh files executable files with Terminal.app)_
	- or with a **Right-click** > **Open with** > **Terminal.app**
	- or with **CLI** from a terminal:

!!! shell
    _Exemple script to rin GeVarLi_
    ```shell
    ./Run_GeVarLi.sh
    ```

!!! warning
   If your reads were generated with an **amplicon protocol**, you will also need to provide the **amplicon primer coordinates** in **BEDPE** format, so the **primers are trimmed appropriately** using **bamclipper**.

!!! tip
   If you don't know the amplicon primer coordinates or don't have coordinates in **BEDPE** format, you can use **cutadapt** to hard-trim the first N nucleotides of your reads (N = yours amplicon primers length). 

3. Yours analyzes will start, with default configuration settings  

!!! note
   Edit **config.yaml** file in ```./configuration/``` directory.  
   Edit **fastq-screen.conf** file in```./configuration/``` directory.

## **Files tree**

```shell
ğŸ§© GeVarLi/
â”‚
â”œâ”€â”€ ğŸ–¥ï¸ï¸ Run_GeVarLi.sh
â”œâ”€â”€ ğŸ“œ Snakefile
â”œâ”€â”€ ğŸ“š README.md
â”œâ”€â”€ ğŸ“š CODE_OF_CONDUCT.md
â”œâ”€â”€ ğŸªª LICENSE.txt
â”œâ”€â”€ ğŸš« .gitignore
â”œâ”€â”€ ğŸ“‚ .git/
â”œâ”€â”€ ğŸ“‚ .snakemake/
â”‚
â”œâ”€â”€ ğŸ“‚ configuration/
â”‚   â”‚
â”‚   â”œâ”€â”€ âš™ï¸ config.yaml
â”‚   â”œâ”€â”€ âš™ï¸ fastq-screen.conf
â”‚   â””â”€â”€ âš™ï¸ multiqc.yaml
â”‚
â”‚
â”œâ”€â”€ ğŸ“‚ workflow/
â”‚   â”‚
â”‚   â”œâ”€â”€ ğŸ“‚ environments/
â”‚   â”‚   â”‚
â”‚   â”‚   â”œâ”€â”€ ğŸœ {TOOL}_v.{VERSION}.yaml
â”‚   â”‚   â””â”€â”€ ğŸœ workflow-base_v.{VERSION}.yaml
â”‚   â”‚
â”‚   â””â”€â”€ ğŸ“‚ snakefiles/
â”‚       â”‚
â”‚       â”œâ”€â”€ ğŸ“œ gevarli.smk
â”‚       â””â”€â”€ ğŸ“œ indexing_genomes.smk
â”‚
â”œâ”€â”€ ğŸ“‚ resources/
â”‚   â”‚
â”‚   â”œâ”€â”€ ğŸ“‚ genomes/
â”‚   â”‚   â”‚
â”‚   â”‚   â”œâ”€â”€ ğŸ§¬ SARS-CoV-2_Wuhan_MN-908947-3.fasta
â”‚   â”‚   â”œâ”€â”€ ğŸ§¬ Monkeypox-virus_Zaire_AF-380138-1.fasta
â”‚   â”‚   â”œâ”€â”€ ğŸ§¬ {REFERENCE}.fasta
â”‚   â”‚   â”œâ”€â”€ ğŸ§¬ QC_Phi-X174_Coliphage_NC-001422-1.fasta
â”‚   â”‚Â   â””â”€â”€ ğŸ§¬ QC_{REFERENCE}.fasta
â”‚   â”‚
â”‚   â”œâ”€â”€ ğŸ“‚ indexes/
â”‚   â”‚   â”‚
â”‚   â”‚   â”œâ”€â”€ ğŸ“‚ minimap2/
â”‚   â”‚   â”‚   â”‚
â”‚   â”‚   â”‚   â””â”€â”€ ğŸ—‚ï¸ {GENOME}.mmi
â”‚   â”‚   â”‚
â”‚   â”‚   â”œâ”€â”€ ğŸ“‚ bwa/
â”‚   â”‚   â”‚   â”‚
â”‚   â”‚   â”‚   â”œâ”€â”€ ğŸ—‚ï¸ {GENOME}.amb
â”‚   â”‚   â”‚   â”œâ”€â”€ ğŸ—‚ï¸ {GENOME}.ann
â”‚   â”‚   â”‚   â”œâ”€â”€ ğŸ—‚ï¸ {GENOME}.bwt
â”‚   â”‚   â”‚   â”œâ”€â”€ ğŸ—‚ï¸ {GENOME}.pac
â”‚   â”‚   â”‚   â””â”€â”€ ğŸ—‚ï¸ {GENOME}.sa
â”‚   â”‚   â”‚
â”‚   â”‚   â””â”€â”€ ğŸ“‚ bowtie2/
â”‚   â”‚       â”‚
â”‚   â”‚       â”œâ”€â”€ ğŸ—‚ï¸ {GENOME}.1.bt2
â”‚   â”‚       â”œâ”€â”€ ğŸ—‚ï¸ {GENOME}.2.bt2
â”‚   â”‚       â”œâ”€â”€ ğŸ—‚ï¸ {GENOME}.3.bt2
â”‚   â”‚       â”œâ”€â”€ ğŸ—‚ï¸ {GENOME}.4.bt2
â”‚   â”‚       â”œâ”€â”€ ğŸ—‚ï¸ {GENOME}.rev.1.bt2
â”‚   â”‚       â””â”€â”€ ğŸ—‚ï¸ {GENOME}.rev.2.bt2
â”‚   â”‚
â”‚   â”œâ”€â”€ ğŸ“‚ nextclade/
â”‚   â”‚   â”‚
â”‚   â”‚   â””â”€â”€ ğŸ“‚ {DATABASE}/
â”‚   â”‚       â”‚
â”‚   â”‚       â”œâ”€â”€ ğŸŒ genemap.gff
â”‚   â”‚       â”œâ”€â”€ ğŸ§ª primers.csv
â”‚   â”‚       â”œâ”€â”€ âœ… qc.json
â”‚   â”‚       â”œâ”€â”€ ğŸ¦  reference.fasta
â”‚   â”‚       â”œâ”€â”€ ğŸ§¬ sequences.fasta
â”‚   â”‚       â”œâ”€â”€ ğŸ·ï¸ tag.json
â”‚   â”‚       â””â”€â”€ ğŸŒ³ tree.json
â”‚   â”‚
â”‚   â”œâ”€â”€ ğŸ“‚ reads/
â”‚   â”‚   â”‚
â”‚   â”‚   â”œâ”€â”€ ğŸ›¡ï¸ .gitkeep
â”‚   â”‚   â”œâ”€â”€ ğŸ“¦ {SAMPLE}_R1.fastq.gz
â”‚   â”‚   â””â”€â”€ ğŸ“¦ {SAMPLE}_R2.fastq.gz
â”‚   â”‚
â”‚   â”œâ”€â”€ ğŸ“‚ primers/
â”‚   â”‚   â”‚
â”‚   â”‚   â”œâ”€â”€ ğŸ“‚ bedpe/
â”‚   â”‚   â”œâ”€â”€ ğŸ“‚ bed/
â”‚   â”‚   â””â”€â”€ ğŸ“‚ fasta/
â”‚   â”‚
â”‚   â”œâ”€â”€ ğŸ“‚ annotations/
â”‚   â”‚   â”‚
â”‚   â”‚   â”œâ”€â”€  SARS-CoV-2_Wuhan_MN-908947-3.bed
â”‚   â”‚   â””â”€â”€  SARS-CoV-2_Wuhan_MN-908947-3.gff3
â”‚   â”‚
â”‚   â””â”€â”€ ğŸ“‚ data_test/
â”‚       â”‚
â”‚       â”œâ”€â”€ ğŸ›¡ï¸ .gitkeep
â”‚       â”œâ”€â”€ ğŸ“¦ SARS-CoV-2_Omicron-BA1_Covid-Seq-Lib-on-MiSeq_250000-reads_R1.fastq.gz
â”‚       â””â”€â”€ ğŸ“¦ SARS-CoV-2_Omicron-BA1_Covid-Seq-Lib-on-MiSeq_250000-reads_R2.fastq.gz
â”‚
â””â”€â”€ ğŸ“‚ documentation/
    â”‚
    â”œâ”€â”€ âš™ï¸ mkdocs.yaml
    â”œâ”€â”€ ğŸŒ extra.css
    â”‚
    â”œâ”€â”€ ğŸ“‚ images/
    â”‚   â”‚
    â”‚   â”œâ”€â”€ ğŸ“ˆ afroscreen.jpeg
    â”‚   â”œâ”€â”€ ğŸ“ˆ download_button.png
    â”‚   â”œâ”€â”€ ğŸ“ˆ GeVarLi_by_DALL-E_icone.png
    â”‚   â”œâ”€â”€ ğŸ“ˆ gevarli_filegraph.png
    â”‚   â””â”€â”€ ğŸ“ˆ gevarli_rulegraph.png
    â”‚
    â”œâ”€â”€ ğŸ“‚ en/
    â”‚   â”‚
    â”‚   â”œâ”€â”€ ğŸ“š index.md
    â”‚   â”‚
    â”‚   â””â”€â”€ ğŸ“‚ pages/
    â”‚       â”‚
    â”‚       â”œâ”€â”€ ğŸ“š acknowledgedments.md
    â”‚       â”œâ”€â”€ ğŸ“š afroscreen
    â”‚       â”œâ”€â”€ ğŸ“š configurations.md
    â”‚       â”œâ”€â”€ ğŸ“š contributions.md
    â”‚       â”œâ”€â”€ ğŸ“š features.md
    â”‚       â”œâ”€â”€ ğŸ“š glossary.md
    â”‚       â”œâ”€â”€ ğŸ“š installations.md
    â”‚       â”œâ”€â”€ ğŸ“š references.md
    â”‚       â”œâ”€â”€ ğŸ“š results.md
    â”‚       â””â”€â”€ ğŸ“š usage.md
    â”‚
    â””â”€â”€ ğŸ“‚ fr/
        â”‚
        â”œâ”€â”€ ğŸ“š index.md
        â”‚
        â””â”€â”€ ğŸ“‚ pages/
            â”‚
            â”œâ”€â”€ ğŸ“š acknowledgedments.md
            â”œâ”€â”€ ğŸ“š afroscreen
            â”œâ”€â”€ ğŸ“š configurations.md
            â”œâ”€â”€ ğŸ“š contributions.md
            â”œâ”€â”€ ğŸ“š features.md
            â”œâ”€â”€ ğŸ“š glossary.md
            â”œâ”€â”€ ğŸ“š installations.md
            â”œâ”€â”€ ğŸ“š references.md
            â”œâ”€â”€ ğŸ“š results.md
            â””â”€â”€ ğŸ“š usage.md
```
