# **RÃ©sultats**

Les rÃ©sultats sont disponible dans le rÃ©pertoire **./results/** .

Ils sont organisÃ©s comme ceci :

- Racine : Les fichiers concaternÃ©s pour tout les Ã©chantillons, 1 par rÃ©fÃ©rence utilisÃ©es

    * les consensus (FASTA)
    * les statistiques de couverture (TSV)
    * les lignÃ©es Nextclade et la qualitÃ© des consensus (TSV)
    * les lignÃ©es Pangolin, si appelÃ© (TSV)

- 00 - ContrÃ´le QualitÃ© : les fichiers de qualitÃ©s sur les lectures brutes (FASTQ)
  * test
  
!!! tip
    Les fichiers Ã©tiquetÃ© comme **[TEMP]** sont supprimÃ©s par dÃ©faut, afin de prÃ©server l'espace disque.  
	Si vous souhaiter les conserver, veuiller Ã©diter le fichier de **configuration**.

## **Arborescence des fichiers**
```shell
ğŸ§© GeVarLi/
â”‚
â””â”€â”€ ğŸ“‚ results/
    â”‚
    â”œâ”€â”€ ğŸŒ All_readsQC_reports.html
    â”œâ”€â”€ ğŸ§¬ All_{REFERENCE}_consensus_sequences.fasta
    â”œâ”€â”€ ğŸ“Š All_{REFERENCE}_genome_coverages.tsv
    â”œâ”€â”€ ğŸ“Š All_{REFERENCE}_nextclade_lineages.tsv
    â”œâ”€â”€ ğŸ“Š All_{REFERENCE}_pangolin_lineages.tsv
    â”‚
    â”œâ”€â”€ ğŸ“‚ 00_Quality_Control/
    â”‚   â”‚
    â”‚   â”œâ”€â”€ ğŸ“‚ fastq-screen/
    â”‚   â”‚   â”‚
    â”‚   â”‚   â”œâ”€â”€ ğŸŒ {SAMPLE}_R{1|2}_screen.html
    â”‚   â”‚   â”œâ”€â”€ ğŸ“ˆ {SAMPLE}_R{1|2}_screen.png
    â”‚   â”‚   â””â”€â”€ ğŸ“„ {SAMPLE}_R{1|2}_screen.txt
    â”‚   â”‚
    â”‚   â”œâ”€â”€ ğŸ“‚ fastqc/
    â”‚   â”‚   â”‚
    â”‚   â”‚   â”œâ”€â”€ ğŸŒ {SAMPLE}_R{1|2}_fastqc.html
    â”‚   â”‚   â””â”€â”€ ğŸ“¦ {SAMPLE}_R{1|2}_fastqc.zip
    â”‚   â”‚
    â”‚   â””â”€â”€ ğŸ“‚ multiqc/
    â”‚       â”‚
    â”‚       â”œâ”€â”€ ğŸŒ multiqc_report.html
    â”‚       â”‚
    â”‚       â””â”€â”€ğŸ“‚ multiqc_data/
    â”‚          â”‚
    â”‚          â”œâ”€â”€ ğŸ“ multiqc.log
    â”‚          â”œâ”€â”€ ğŸ“„ multiqc_citations.txt
    â”‚          â”œâ”€â”€ ğŸŒ€ multiqc_data.json
    â”‚          â”œâ”€â”€ ğŸ“„ multiqc_fastq_screen.txt
    â”‚          â”œâ”€â”€ ğŸ“„ multiqc_fastqc.txt
    â”‚          â”œâ”€â”€ ğŸ“„ multiqc_general_stats.txt
    â”‚          â””â”€â”€ ğŸ“„ multiqc_sources.txt
    â”‚
    â”œâ”€â”€ ğŸ“‚ 01_Trimming
    â”‚   â”‚
    â”‚   â”œâ”€â”€ ğŸ“‚ cutadapt/
    â”‚   â”‚   â”‚
    â”‚   â”‚   â””â”€â”€ ğŸ“¦ {SAMPLE}_cutadapt-removed_R{1|2}.fastq.gz #[TEMP]
    â”‚   â”‚
    â”‚   â””â”€â”€ ğŸ“‚ sickle/
    â”‚       â”‚
    â”‚       â”œâ”€â”€ ğŸ“¦ {SAMPLE}_sickle-trimmed_R{1|2}.fastq.gz   #[TEMP]
    â”‚       â””â”€â”€ ğŸ“¦ {SAMPLE}_sickle-trimmed_SE.fastq.gz       #[TEMP]
    â”‚
    â”œâ”€â”€ ğŸ“‚ 02_Mapping/
    â”‚   â”‚
    â”‚   â”œâ”€â”€ ğŸ§­ {SAMPLE}_{REFERENCE}_{ALIGNER}_mark-dup.bam
    â”‚   â”œâ”€â”€ ğŸ—‚ï¸ {SAMPLE}_{REFERENCE}_{ALIGNER}_mark-dup.bam.bai
    â”‚   â”œâ”€â”€ ğŸ§­ {SAMPLE}_{REFERENCE}_{ALIGNER}_mark-dup.primerclipped.bam
    â”‚   â”œâ”€â”€ ğŸ—‚ï¸ {SAMPLE}_{REFERENCE}_{ALIGNER}_mark-dup.primerclipped.bam.bai
    â”‚   â”œâ”€â”€ ğŸ§­ {SAMPLE}_{REFERENCE}_{ALIGNER}_mapped.sam                     #[TEMP]
    â”‚   â”œâ”€â”€ ğŸ§­ {SAMPLE}_{REFERENCE}_{ALIGNER}_sorted-by-names.bam            #[TEMP]
    â”‚   â”œâ”€â”€ ğŸ§­ {SAMPLE}_{REFERENCE}_{ALIGNER}_fixed-mate.bam                 #[TEMP]
    â”‚   â””â”€â”€ ğŸ§­ {SAMPLE}_{REFERENCE}_{ALIGNER}_sorted.bam                     #[TEMP]
    â”‚
    â”œâ”€â”€ ğŸ“‚ 03_Coverage/
    â”‚   â”‚
    â”‚   â”œâ”€â”€ ğŸ“Š {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_coverage-stats.tsv
    â”‚   â”œâ”€â”€ ğŸ›ï¸ {SAMPLE}_{REFERENCE}_{ALIGNER}_genome-cov.bed              #[TEMP]
    â”‚   â”œâ”€â”€ ğŸ›ï¸ {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_min-cov-filt.bed   #[TEMP]
    â”‚   â””â”€â”€ ğŸ›ï¸ {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_low-cov-mask.bed   #[TEMP]
    â”‚
    â”œâ”€â”€ ğŸ“‚ 04_Variants/
    â”‚   â”‚
    â”‚   â”œâ”€â”€ ğŸ§¬ {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_masked-ref.fasta
    â”‚   â”œâ”€â”€ ğŸ—‚ï¸ {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_masked-ref.fasta.fai
    â”‚   â”œâ”€â”€ ğŸ§­ {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_indel-qual.bam
    â”‚   â”œâ”€â”€ ğŸ—‚ï¸ {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_indel-qual.bai
    â”‚   â”œâ”€â”€ ğŸ§®ï¸ {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_variant-call.vcf
    â”‚   â”œâ”€â”€ ğŸ§®ï¸ {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_variant-filt.vcf
    â”‚   â”œâ”€â”€ ğŸ“¦ {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_variant-filt.vcf.bgz     #[TEMP]
    â”‚   â””â”€â”€ ğŸ—‚ï¸ {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_variant-filt.vcf.bgz.tbi #[TEMP]
    â”‚
    â”œâ”€â”€ ğŸ“‚ 05_Consensus/
    â”‚   â”‚
    â”‚   â”œâ”€â”€ ğŸ§¬ {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_consensus.fasta
    â”‚   â”‚
    â”‚   â””â”€â”€ ğŸ“‚ ivar_consensus-quality/
    â”‚       â”‚
    â”‚       â””â”€â”€ ğŸŒ {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_ivar_consensus.qual
    â”‚
    â”œâ”€â”€ ğŸ“‚ 06_Lineages/
    â”‚   â”‚
    â”‚   â”œâ”€â”€ ğŸ“Š {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_nextclade-report.tsv
    â”‚   â”œâ”€â”€ ğŸ“Š {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_pangolin-report.csv
    â”‚   â”‚
    â”‚   â””â”€â”€ ğŸ“‚ {SAMPLE}_{REFERENCE}_{ALIGNER}_{MINCOV}_nextclade-all/
    â”‚       â”‚
    â”‚       â”œâ”€â”€ ğŸ§¬ nextclade.aligned.fasta
    â”‚       â”œâ”€â”€ ğŸ“Š nextclade.csv
    â”‚       â”œâ”€â”€ ğŸ“Š nextclade.errors.csv
    â”‚       â”œâ”€â”€ ğŸ“Š nextclade.insertions.csv
    â”‚       â”œâ”€â”€ ğŸŒ€ nextclade.json
    â”‚       â”œâ”€â”€ ğŸŒ€ nextclade.ndjson
    â”‚       â”œâ”€â”€ ğŸŒ€ nextclade.auspice.json
    â”‚       â””â”€â”€ ğŸ§¬ nextclade_{GENE}.translation.fasta
    â”‚
    â””â”€â”€ ğŸ“‚ 10_Reports/
        â”‚
        â”œâ”€â”€ âš™ï¸ config.log
        â”œâ”€â”€ ğŸ“ settings.log
        â”œâ”€â”€ ğŸœ gevarli-base_v.{VERSION}.yaml
        â”‚
        â”œâ”€â”€ ğŸ“‚ conda_env/
        â”‚   â”‚
        â”‚   â””â”€â”€ ğŸ“„ {TOOLS}_v.{version}.yaml
        â”‚
        â”œâ”€â”€ ğŸ“‚ files-summaries
        â”‚   â”‚
        â”‚   â””â”€â”€ ğŸ“„ {PIPELINE}_files-summary.txt
        â”‚
        â”œâ”€â”€ ğŸ“‚ graphs/
        â”‚   â”‚
        â”‚   â”œâ”€â”€ ğŸ“ˆ {PIPELINE}_dag.{PNG/PDF}
        â”‚   â”œâ”€â”€ ğŸ“ˆ {PIPELINE}_filegraph.{PNG/PDF}
        â”‚   â””â”€â”€ ğŸ“ˆ {PIPELINE}_rulegraph.{PNG/PDF}
        â”‚
        â””â”€â”€ ğŸ“‚ tools-log/
            â”‚
            â””â”€â”€ ğŸ“‚ {TOOL}/
                â”‚
                â””â”€â”€ ğŸ“ {SAMPLE}_{REFERENCE}_{ALIGNER}.log
```
